cosmo/clients/netbox_v4.py

```diff
@@ -0,0 +1,327 @@
+                    import_targets {
+                      name
+                    }
+                }
+            }
+        ''')
+
+        return self.client.query(query_template.substitute())['data']
+
+    def _merge_into(self, data: dict, query_data):
+        return {
+            **data,
+            **query_data,
+        }
+
+
+class StaticRouteQuery(ParallelQuery):
+
+    def _fetch_data(self, kwargs):
+        device_list = kwargs.get("device_list")
+        return self.client.query_rest("api/plugins/routing/staticroutes/", {"device": device_list})
+
+    def _merge_into(self, data: dict, query_data):
+        for d in data['device_list']:
+            device_static_routes = list(filter(lambda sr: str(sr['device']['id']) == d['id'], query_data))
+            d['staticroute_set'] = device_static_routes
+
+        return data
+
+
+class DeviceDataQuery(ParallelQuery):
+    def _fetch_data(self, kwargs):
+        device = kwargs.get("device")
+        query_template = Template(
+            """
+            query {
+              device_list(filters: {
+                name: { i_exact: $device },
+              }) {
+                id
+                name
+                serial
+
+                device_type {
+                  slug
+                }
+                platform {
+                  manufacturer {
+                    slug
+                  }
+                  slug
+                }
+                primary_ip4 {
+                  address
+                }
+
+                interfaces {
+                  id
+                  name
+                  enabled
+                  type
+                  mode
+                  mtu
+                  mac_address
+                  description
+                  vrf {
+                    id
+                  }
+                  lag {
+                    id
+                  }
+                  ip_addresses {
+                    address
+                  }
+                  untagged_vlan {
+                    id
+                    name
+                    vid
+                  }
+                  tagged_vlans {
+                    id
+                    name
+                    vid
+                  }
+                  tags {
+                    name
+                    slug
+                  }
+                  parent {
+                    id
+                  }
+                  custom_fields
+                }
+              }
+            }"""
+        )
+
+        query = query_template.substitute(
+            device=json.dumps(device)
+        )
+        return self.client.query(query)['data']
+
+    def _merge_into(self, data: dict, query_data):
+        if 'device_list' not in data:
+            data['device_list'] = []
+
+        data['device_list'].extend(query_data['device_list'])
+
+        return data
+
+
+class NetboxV4Strategy:
+
+    def __init__(self, url, token):
+        self.client = NetboxAPIClient(url, token)
```

Comment:
I feel like maybe we could use subclass and super() here. would be cleaner imo. as there's no need for dynamic client type injection. but not sure, it's up to you.

Reply:
I specifically moved it out of a base class to be able to pass it into the queries without having the Queries inherit the scope of the entire Client abomination.